### PR TITLE
2.28 backport - Fix armclang compile fail

### PR DESCRIPTION
**Pull request by daverodgman**
## Description

Backport (not trivial) of #7651 .

## PR checklist

- [x] **changelog** provided
- [x] **backport** this is it
- [x] **tests** additional tests added

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
There seems to have been some mixup between Jenkins and GitHub with the reporting of CI results for #7683 and #7684. The CI for this pull request is running at [openci](https://ci.staging.trustedfirmware.org/blue/organizations/jenkins/mbed-tls-pr-head/detail/PR-7684-head/1/pipeline/) (passed) and [internal ci](https://jenkins-mbedtls.oss.arm.com/blue/organizations/jenkins/mbed-tls-pr-head/detail/PR-7684-head/1/pipeline/) (still running).

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM

File: ChangeLog.d/armclang-compile-fix.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Bugfix
+   * Fix clang and armclang compilation error when targeting certain Arm
+     M-class CPUs (Cortex-M0, Cortex-M0+, Cortex-M1, Cortex-M23,
+     SecurCore SC000). Fixes #1077.
+Changes
+   * Enable Arm / Thumb bignum assembly for most Arm platforms when
```
[the remark this answers is not in the document]
Comment (daverodgman) [reply ChangeLog.d/armclang-compile-fix.txt]: I think that's probably sensible. I've pushed a minimal fix (disable Thumb 1 asm if not on GCC), plus the extra tests.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM (reviewed not as a backport, but as a standalone change to an LTS branch)

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Pull request merged**